include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    constexpr aligned_sequence_adaptor_constant_access & operator=(aligned_sequence_adaptor_constant_access && rhs)
+    {
+        gap_vector = std::move(rhs.gap_vector);
+        sequence = std::move(rhs.sequence);
+        rs.set_vector(&gap_vector);
+        ss.set_vector(&gap_vector);
+        return *this;
+    }
+
+    //!\brief Use default deconstructor.
+    ~aligned_sequence_adaptor_constant_access() = default;
+    //!\}
+
+    //!\brief
+    /*!\name Constructors of sequence concept.
+    * \{
+    */
+    //!\brief Construct by single value repeated 'size' times.
+    constexpr aligned_sequence_adaptor_constant_access(size_type size, value_type value)
+    {
+        assign(size, value);
+    };
+
+    //!\brief Construct by range given by two iterators of an aligned sequence.
+    constexpr aligned_sequence_adaptor_constant_access(iterator it1, iterator it2)
+    {
+        assign(it1, it2);
+    };
+
+    //!\brief Construct by sequence.
+    constexpr aligned_sequence_adaptor_constant_access(inner_type sequence_)
+    {
+        assign(sequence_);
+    };
+
+    //!\brief Construct by initializer_list for inner sequence container.
+    constexpr aligned_sequence_adaptor_constant_access(std::initializer_list<value_type> l)
+    {
+        assign(l);
+    };
+    //!\}
+
+    /*!\name Iterators
+     * \{
+    */
+    //!\brief Return iterator pointing to first element of underlying sequence.
+    auto begin() noexcept
+    {
+        return iterator{*this, 0};
+    }
+
+    //!\brief Return iterator pointing to past-the-end element of gapped sequence.
+    auto end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\brief Return const iterator pointing to first element of underlying sequence
+    const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this, 0};
+    }
+
+    //!\brief Return const iterator pointing to past-the-end element of underlying sequence.
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+     //!\}
+
+    /*!\name Boolean operators
+     * \{
+     */
+    /*!\brief Equality operator for aligned sequences.
+    *
+    * Two aligned sequences are the same if their literal sequences and gap
+    * positions are the same.
+    */
+    bool operator==(aligned_sequence_t const & rhs) const
+    {
+        return sequence == rhs.sequence && gap_vector == rhs.gap_vector;
+    }
+
+    //!\brief Unequality operator for aligned sequences.
+    bool operator!=(aligned_sequence_t const & rhs) const
+    {
+        return !(*this == rhs);
+    }
+
+    //!\brief Swap two aligned sequences and their support structures.
+    void swap(aligned_sequence_t & rhs)
+    {
+        sequence.swap(rhs.sequence);
+        std::swap(gap_vector, rhs.gap_vector);
+        rs.set_vector(&gap_vector);
+        rhs.rs.set_vector(&rhs.gap_vector);
+        ss.set_vector(&gap_vector);
+        rhs.ss.set_vector(&gap_vector);
```

Comment:
can the support data structures not be swapped?

Reply:
Ah, no probably not.

Reply:
Yeah, this was issue https://github.com/xxsds/sdsl-lite/issues/15. The support structures need the new address of the bit_vector which is done by set_vector.